modules/RTC/TraceablePeerConnection.js

```diff
@@ -910,68 +1673,148 @@ TraceablePeerConnection.prototype.createAnswer
-                    answer.sdp = this.rtxModifier.modifyRtxSsrcs(answer.sdp);
+                    resultSdp.sdp
+                        = this.rtxModifier.modifyRtxSsrcs(resultSdp.sdp);
                     this.trace(
-                        'createAnswerOnSuccess::postTransform (rtx modifier)',
-                        dumpSDP(answer));
+                        `create${logName}`
+                             + 'OnSuccess::postTransform (rtx modifier)',
+                        dumpSDP(resultSdp));
                 }
 
                 // Fix the setup attribute (see _fixAnswerRFC4145Setup for
                 //  details)
-                const remoteDescription = new SDP(this.remoteDescription.sdp);
-                const localDescription = new SDP(answer.sdp);
+                if (!isOffer) {
+                    const remoteDescription
+                        = new SDP(this.remoteDescription.sdp);
+                    const localDescription = new SDP(resultSdp.sdp);
 
-                _fixAnswerRFC4145Setup(remoteDescription, localDescription);
-                answer.sdp = localDescription.raw;
+                    _fixAnswerRFC4145Setup(remoteDescription, localDescription);
+                    resultSdp.sdp = localDescription.raw;
+                }
+
+                const ssrcMap = extractSSRCMap(resultSdp);
+
+                logger.info('Got SSRC MAP: ', ssrcMap);
 
-                this.eventEmitter.emit(RTCEvents.SENDRECV_STREAMS_CHANGED,
-                    extractSSRCMap(answer));
+                // Set up the ssrcHandler for the new track before we add it at
+                // the lower levels
+                this._applyLocalSSRCMap(ssrcMap);
 
-                successCallback(answer);
+                successCallback(resultSdp);
             } catch (e) {
-                this.trace('createAnswerOnError', e);
-                this.trace('createAnswerOnError', dumpSDP(answer));
-                logger.error('createAnswerOnError', e, dumpSDP(answer));
+                this.trace(`create${logName}OnError`, e);
+                this.trace(`create${logName}OnError`, dumpSDP(resultSdp));
+                logger.error(
+                    `create${logName}OnError`, e, dumpSDP(resultSdp));
                 failureCallback(e);
             }
         },
         err => {
-            this.trace('createAnswerOnFailure', err);
-            this.eventEmitter.emit(RTCEvents.CREATE_ANSWER_FAILED, err,
-                this.peerconnection);
+            this.trace(`create${logName}OnFailure`, err);
+            const eventType
+                = isOffer
+                    ? RTCEvents.CREATE_OFFER_FAILED
+                    : RTCEvents.CREATE_ANSWER_FAILED;
+
+            this.eventEmitter.emit(eventType, err, this.peerconnection);
             failureCallback(err);
         },
         constraints);
-    };
+};
 
-TraceablePeerConnection.prototype.addIceCandidate
+/* eslint-enable max-params */
+
+/**
+ *
```

Comment:
some extra docs would help here. Also, what SSRC are we extracting here? Is it the primary one? Then how about extractPrimerySsrc?